src/main/java/org/students/AsymmetricCryptographyService.java

```diff
@@ -0,0 +1,66 @@
+package org.students;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.ObjectOutputStream;
+import java.io.Serializable;
+import java.security.KeyPair;
+
+public abstract class AsymmetricCryptographyService {
+    public abstract byte[] digestObject(Serializable messageObject);
+    // - digests the serializable object and returns a byte[] containing the hash
```

Comment:
Please use Javadoc notation for commenting public methods. The reason is that IDEs can parse Javadoc comments and thus allow quick overview of method's usage without having to read the implementation in detail. Also Javadoc can be used to generate documentation for the project.